clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,431 @@
+        Ok(_) => text_response(StatusCode::OK, "Application stopped successfully."),
+        Err(error) => text_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            format!("Unable to stop application: {error}"),
+        ),
+    }
+}
+
+fn update_config(body: &str) -> Response {
+    let mut config: TeliodDaemonConfig = match fs::read_to_string(TELIOD_CFG)
+        .and_then(|content| serde_json::from_str(&content).map_err(|e| e.into()))
+    {
+        Ok(config) => config,
+        Err(e) => {
+            eprintln!("Error reading config file: {}", e);
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to read existing config",
+            );
+        }
+    };
+    let updated_config = match serde_json::from_str::<TeliodDaemonConfigPartial>(body) {
+        Ok(updates) => updates,
+        Err(e) => {
+            eprintln!("Error parsing config json: {}", e);
+            return text_response(StatusCode::BAD_REQUEST, "Invalid JSON payload");
+        }
+    };
+
+    config.update(updated_config);
+
+    match fs::write(TELIOD_CFG, serde_json::to_string_pretty(&config).unwrap()) {
+        Ok(_) => text_response(StatusCode::OK, "Configuration updated successfully"),
+        Err(_) => text_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Failed to write updated config",
+        ),
+    }
+}
+
+fn get_status() -> Response {
+    if !is_teliod_running() {
+        return text_response(StatusCode::GONE, "Application is not running.");
+    }
+
+    match teliod_blocking_query!(ClientCmd::GetStatus) {
+        Ok(Ok(daemon_reply)) => match CommandResponse::deserialize(&daemon_reply) {
+            Ok(CommandResponse::StatusReport(status)) => text_response(
+                StatusCode::OK,
+                serde_json::to_string_pretty(&status).unwrap_or_default(),
+            ),
+            Ok(cmd_response) => text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Unexpected Teliod response: {}", cmd_response.serialize()),
+            ),
+            Err(error) => text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to retrieve status: {}", error),
+            ),
+        },
+        Ok(Err(error)) => text_response(
+            StatusCode::GONE,
+            format!("Failed to communicate with the daemon: {}", error),
+        ),
+        Err(error) => text_response(
+            StatusCode::GATEWAY_TIMEOUT,
+            format!("Failed to communicate with the daemon: {}", error),
+        ),
+    }
+}
+
+fn get_teliod_logs() -> Response {
+    match fs::read_to_string(TELIOD_LOG) {
+        Ok(logs) => text_response(StatusCode::OK, logs),
+        Err(error) => text_response(
+            StatusCode::BAD_GATEWAY,
+            format!("Error reading teliod log file: {}", error),
+        ),
+    }
+}
+
+fn get_meshnet_logs() -> Response {
```

Comment:
Is there a good reason to separate `teliod` and `meshnet` logs? maybe it would be worth to interleave them, like apps do? In that case it is slightly easier to get them from users, as there is single file instead of two.

Reply:
In my POV I'd prefer to keep them separated simply because meshnet logs will bury `teliod` logs which also convey a different category of information.

Reply:
ok